test/connect.spec.tsx

```diff
@@ -105,38 +107,91 @@ describe('connect', () => {
       expect(instance.myText).to.equal('baz');
     });
 
-    it('maps actions on creation', () => {
+    it('maps action dispatcher to callback on creation', () => {
       instance.callback('foo');
 
       expect(actions).to.deep.equal([{type: 'foo'}]);
     });
 
+    it('maps action creator to callback on creation', () => {
+      @connect<CustomComponent2, RootState, Action>(
+        null,
+        {callback: (type: 'foo' | 'bar') => ({type})}
+      )
+      @component
+      class CustomComponent2 extends Composite {
+        @property callback: (type: string) => any;
+      }
+
+      (<CustomComponent2/> as CustomComponent2).callback('foo');
+
+      expect(actions).to.deep.equal([{type: 'foo'}]);
+    });
+
+    it('maps action dispatcher to event on creation', () => {
+      @connect<CustomComponent2, RootState, Action>(
+        null,
+        dispatch => ({onCallback: ev => dispatch({type: ev.payload})})
+      )
+      @component
+      class CustomComponent2 extends Composite {
+        @event onCallback: Listeners<{target: CustomComponent2, payload: 'foo' | 'bar'}>;
+      }
+
+      (<CustomComponent2/> as CustomComponent2).onCallback.trigger({payload: 'foo'});
+
+      expect(actions).to.deep.equal([{type: 'foo'}]);
+    });
+
+    it('maps action creator to event on creation', () => {
+      @connect<CustomComponent2, RootState, Action>(
+        null,
+        {onCallback: ev => ({type: ev.payload})}
+      )
+      @component
+      class CustomComponent2 extends Composite {
+        @event onCallback: Listeners<{target: CustomComponent2, payload: 'foo' | 'bar'}>;
+      }
+
+      (<CustomComponent2/> as CustomComponent2).onCallback.trigger({payload: 'foo'});
+
+      expect(actions).to.deep.equal([{type: 'foo'}]);
+    });
+
```

Comment:
Does it support to use `connect` method to only map `state` but not `dispatch` like you provide a `null` value to state? I think this could be beneficial for Components that only need to read from the state. Can we add a unit test for such cases?

Reply:
There is no explicit test, but one at line 178. I can add one in an additonal commit.